src/libsystemd-network/ndisc-option.c

```diff
@@ -88,6 +89,13 @@ static void ndisc_dnssl_done(sd_ndisc_dnssl *dnssl) {
         strv_free(dnssl->domains);
 }
 
+static void ndisc_dnr_done(sd_ndisc_dnr *dnr) {
+        if (!dnr)
```

Comment:
our deep destructors (usually called xyz_free()) (and the refcounted version of it, i.e. xyz_unref()) are typically written to handle NULL pointers gracefully, but our shallow destructors (usually called xyz_done()) should hit an assert() instead.

Reply:
I'm going to leave this as is so it matches the surrounding functions, then maybe it can be renamed or something in the future.

Reply:
I'm going to leave this as is so it matches the surrounding functions. Then maybe they can all be changed in a future cleanup.

Reply:
ok